src/languages/yaml.js

```diff
@@ -46,6 +46,19 @@ export default function(hljs) {
       TEMPLATE_VARIABLES
     ]
   };
+  var INSIDE_OBJECT_STRING = {
+    className: 'string',
+    relevance: 0,
+    variants: [
+      {begin: /'/, end: /'/},
+      {begin: /"/, end: /"/},
+      {begin: /\S*[^\s,\]}]/}
```

Comment:
`{[` and are off limits too IIRC.

Reply:
Perhaps this should be \S+ with a look-ahead instead? Would that more accurately express the intent?

Reply:
I don't think a look-ahead is what we want here, since it doesn't matter what is immediately before or after the string, it just can't start/end with a comma or brace or bracket.

Reply:
>  it just can't start/end with a comma or brace or bracket.

That's exactly what is "after the string"...

```
\S+(?=[\s,\]}])
```

Plus now the negation is removed which is easier to follow.